### PR TITLE
DisplaySettings: Add theme selection and preview

### PR DESCRIPTION
**Pull request by MacDue**
Screenshot from 2022-04-02 03-03-19

This PR adds a Windows XP-ish inspired theme selector + preview to the display settings.
I thought this may possibly replace the theme selection on the 'Serenity Menu', though I've not included that change in this PR.

### CODE REVIEW

**Comment by AtkinsSJ** [comment]
Is the new widget the same as the ThemeEditor one? If so, it would be good to move it into a shared library instead of duplicating it. (Probably LibGUI)

**Comment by MacDue** [comment]
It's not the same widget, but they're probably similar enough that most of the code could be moved to some shared base widget

**Comment by MacDue** [comment]
Now the shared code from the ThemeEditor preview has been moved to a common AbstactThemePreview widget base class.

File: Userland/Applications/DisplaySettings/main.cpp [hunk 1]
```diff
@@ -27,6 +28,7 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     auto app_icon = GUI::Icon::default_icon("app-display-settings");
 
     auto window = TRY(GUI::SettingsWindow::create("Display Settings"));
+    (void)TRY(window->add_tab<DisplaySettings::ThemesSettingsWidget>("Themes"));
```
Comment (awesomekling) [inline Userland/Applications/DisplaySettings/main.cpp:31]: Please put the themes tab after the background tab :^)

Comment (MacDue) [reply Userland/Applications/DisplaySettings/main.cpp]: Done :)

**Pull request merged**